### PR TITLE
10.0 mig account invoice supplierinfo update

### PR DESCRIPTION
**Pull request by mourad-ehm**

### CODE REVIEW

File: account_invoice_supplierinfo_update/README.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=====================================================
+Update Supplier Info of product from Supplier Invoice
+=====================================================
+
+This module allows in the supplier invoice, to automatically update all
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/README.rst]: "This module allows to automatically update all products information in vendor bill for which the purchase information on the line are different from the supplier information defined in the product form."

File: account_invoice_supplierinfo_update/README.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+products informations whose purchase informations on the line are different
+from the supplier informations, defined on the product form.
+
+It creates a new supplier information line if there is not or it updates the
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/README.rst]: It creates a new supplier information line if there is not any or it updates the
first one in the list.

File: account_invoice_supplierinfo_update/README.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+first.
+
+This module add an extra button 'Check Supplier Informations' on supplier
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/README.rst]: This module adds an new button 'Check Supplier Informations' in supplier

File: account_invoice_supplierinfo_update/README.rst [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+invoice form.
+
+.. image:: /account_invoice_supplierinfo_update/static/description/supplier_invoice_form.png
+
+
+When user click on it, he can see supplier informations changes, that will
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/README.rst]: When the user clicks on it, he can see the supplier information changes that will
apply. Optionally, he can remove some temporary changes, specially, if,
for example, a supplier applied an exceptional price change.

File: account_invoice_supplierinfo_update/README.rst [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+be applied. Optionnaly, he can unselect some temporary changes, specially, if
+supplier applied an exceptionnal price change.
+
+.. image:: /account_invoice_supplierinfo_update/static/description/main_screenshot.png
+
+* blue: Create a full new supplier info
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/README.rst]: Creates a full new supplier info line

File: account_invoice_supplierinfo_update/README.rst [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+* brown: Adding a partnerinfo line in a existing supplierinfo
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/README.rst]: Adds a partnerinfo line in a existing supplierinfo one

File: account_invoice_supplierinfo_update/README.rst [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+* black: Updating a current setting, displaying price variation (%)
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/README.rst]: Updates current settings, displaying price variation (%)

File: account_invoice_supplierinfo_update/README.rst [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+This module add an extra boolean field 'Supplier Informations Checked' in the
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/README.rst]: This module adds an extra boolean field 'Supplier Informations Checked' in the
'Other Info' tab inside the supplier invoice form, that indicates that the prices
have been checked and supplierinfo updated (or eventually that the 
changes have been ignored).

File: account_invoice_supplierinfo_update/README.rst [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+=====
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/142/8.0
+
+Bug Tracker
+===========
+
+Bugs are tracked on GitHub Issues. In case of trouble, please check there
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/README.rst]: please use the usual template including the URL

File: account_invoice_supplierinfo_update/README.rst [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+if your issue has already been reported. If you spotted it first,
+help us smashing it by providing a detailed and welcomed feedback.
+
+Known Issues / Roadmap
+======================
+
+* This module does not manage correctly difference if
+    * invoice line UoM are not the same as Supplierinfo UoM
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/README.rst]: add blank line before the sublist

File: account_invoice_supplierinfo_update/README.rst [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+Maintainer
+----------
+ 
+.. image:: https://odoo-community.org/logo.png
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose mission is to support the collaborative development of Odoo features and promote its  widespread use.
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/README.rst]: update to newest template

File: account_invoice_supplierinfo_update/__manifest__.py [hunk 12]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# © 2016 Chafique DELLI @ Akretion
+# Copyright (C) 2016-Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Account Invoice - Supplier Info Update',
+    'summary': 'In the supplier invoice, automatically update all products '
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/__manifest__.py]: updates

File: account_invoice_supplierinfo_update/demo/account_invoice.xml [hunk 13]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+
+    <!-- <record id="account_invoice_1" model="account.invoice">
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/demo/account_invoice.xml]: remove the file if empty

File: account_invoice_supplierinfo_update/demo/res_groups.xml [hunk 14]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+<!--
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/demo/res_groups.xml]: idem

File: account_invoice_supplierinfo_update/views/account_invoice_view.xml [hunk 15]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+    <record id="view_account_invoice_form" model="ir.ui.view">
+        <field name="model">account.invoice</field>
+        <field name="inherit_id" ref="account.invoice_supplier_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//button[@name='action_invoice_open']" position="after">
+                <button name="check_supplierinfo" string="Check Supplier Infos"
+                        help="Check if the invoice informations are similar from the supplier informations"
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/views/account_invoice_view.xml]: information (never plural)
Why define the help message here: wouldnot it be better at the model definition?

File: account_invoice_supplierinfo_update/wizard/wizard_update_invoice_supplierinfo.xml [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+                    <button name="update_supplierinfo"
+                            string="Update without Validating"
+                            type="object"
+                            class="oe_highlight"/>
+                    <button name="update_supplierinfo_validate"
+                            string="Update and Validate"
+                            type="object" states="draft,proforma2"
+                            groups="account.group_account_invoice"/>
+                    <button name="set_supplierinfo_ok" string="Ignore" type="object"
+                        confirm="This will mark the invoice as checked. Are you sure? It will be possible to uncheck manually in the 'Other Info' tab." help="This will mark the invoice as checked. It will be possible to uncheck manually in the 'Other Info' tab."/>
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/wizard/wizard_update_invoice_supplierinfo.xml]: the confirm string seems very long...

File: account_invoice_supplierinfo_update/wizard/wizard_update_invoice_supplierinfo_line.py [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+
+    state = fields.Selection(selection=_SELECTION_STATE)
+
+    @api.depends('current_price', 'new_price')
+    @api.multi
+    def _compute_price_variation(self):
+        self.write({'price_variation': False})
+        for line in self.filtered('current_price'):
+            line.price_variation = 100 *\
+                (line.new_price - line.current_price) / line.current_price
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/wizard/wizard_update_invoice_supplierinfo_line.py:54]: are you sure the line.current_price is not 0

File: account_invoice_supplierinfo_update/models/account_invoice_line.py [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+        price_variation = False
+
+        if not supplierinfo:
+            state = 'new_supplierinfo'
+        else:
+            state = 'update_supplierinfo'
+            # Compute price variation
+            if supplierinfo.price:
+                price_variation = 100 *\
+                    (self.price_unit - supplierinfo.price) / supplierinfo.price
```
Comment (elicoidal) [inline account_invoice_supplierinfo_update/models/account_invoice_line.py:43]: Are you sure that supplierinfo.price is not 0

**Review by elicoidal — CHANGES_REQUESTED** [review]
thanks for your contribution!

File: account_invoice_supplierinfo_update/wizard/wizard_update_invoice_supplierinfo_line.py (hunk 17 above)
Comment (mourad-ehm) [reply account_invoice_supplierinfo_update/wizard/wizard_update_invoice_supplierinfo_line.py:54]: Yes, I test it self.filtered('current_price') eliminates object with line.current_price =0

File: account_invoice_supplierinfo_update/models/account_invoice_line.py (hunk 18 above)
Comment (mourad-ehm) [reply account_invoice_supplierinfo_update/models/account_invoice_line.py:43]: Yes, "if supplierinfo.price (price=0.0)" is eq to "if False". (tested)

**Comment by mourad-ehm** [comment]
thanks @elicoidal for review. I fixed your remarks.

File: account_invoice_supplierinfo_update/wizard/wizard_update_invoice_supplierinfo_line.py [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+        digits=dp.get_precision('Product Price'),
+        readonly=True)
+
+    new_price = fields.Float(
+        string='New Unit Price', digits=dp.get_precision('Product Price'),
+        required=True)
+
+    price_variation = fields.Float(
+        string='Price Variation (%)', compute='_compute_price_variation',
+        digits_compute=dp.get_precision('Discount'))
```
Comment (EBII) [inline account_invoice_supplierinfo_update/wizard/wizard_update_invoice_supplierinfo_line.py]: Warning on install 
parameter 'digits_compute' is no longer supported; use 'digits' instead.

**Review by EBII — CHANGES_REQUESTED** [review]

**Review by elicoidal — APPROVED** [review]

File: account_invoice_supplierinfo_update/wizard/wizard_update_invoice_supplierinfo_line.py (hunk 19 above)
Comment (mourad-ehm) [reply account_invoice_supplierinfo_update/wizard/wizard_update_invoice_supplierinfo_line.py]: I fixed it. Thanks for review

**Review by legalsylvain — APPROVED** [review]
:+1: code review & Test. 
Thanks a lot for adapting this module to the new supplierinfo model. (More simple by the way, I like it !)

**Comment by mourad-ehm** [comment]
Hi @jgrandguillaume, could you merge this PR please ?

**Review by StefanRijnhart — APPROVED** [review]
Tested and code reviewed. Thanks!

**Pull request merged**